proxy/envoy/config.go

```diff
@@ -658,23 +658,27 @@ func appendPortToDomains(domains []string, port int) []string {
 	return domainsWithPorts
 }
 
-func buildEgressFromSidecarVirtualHostOnPort(rule *proxyconfig.EgressRule, ruleKey string,
+func buildEgressFromSidecarVirtualHostOnPort(rule *proxyconfig.EgressRule,
 	mesh *proxyconfig.ProxyMeshConfig, port *model.Port) *VirtualHost {
 	var externalTrafficCluster *Cluster
 
 	protocolToHandle := port.Protocol
 	if protocolToHandle == model.ProtocolGRPC {
 		protocolToHandle = model.ProtocolHTTP2
 	}
-	protocolSuffix := "-" + strings.ToLower(string(protocolToHandle))
+
+	domain := rule.Destination.Service
+
+	// the name of the virtual host and of the original dst cluster
+	name := domain + ":" + strconv.Itoa(port.Port)
 
 	if rule.UseEgressProxy {
 		externalTrafficCluster = buildOutboundCluster("istio-egress", port, nil)
 		externalTrafficCluster.ServiceName = ""
 		externalTrafficCluster.Type = ClusterTypeStrictDNS
 		externalTrafficCluster.Hosts = []Host{{URL: fmt.Sprintf("tcp://%s", mesh.EgressProxyAddress)}}
 	} else {
-		externalTrafficCluster = buildOriginalDSTCluster("orig-dst-cluster"+protocolSuffix, mesh.ConnectTimeout)
```

Comment:
Please keep this if possible. We need to deduplicate the number of orig_dst_clusters